source/features/fault.cc

```diff
@@ -348,7 +404,9 @@ namespace WorldBuilder
               );
 
       // todo: explain and check -starting_depth
-      if (depth <= maximum_depth && depth >= starting_depth && depth <= maximum_total_fault_length + maximum_fault_thickness)
+      if (surface_bounding_box.point_inside(Point<2>(position_in_natural_coordinates.get_surface_coordinates(),
+                                                     world->parameters.coordinate_system->natural_coordinate_system())) &&
+          depth <= maximum_depth && depth >= starting_depth && depth <= maximum_total_fault_length + maximum_fault_thickness)
```

Comment:
Can you turn this around? With `&&` it should directly return false as soon as one of the conditions is false. So doing the cheap depth checks should be cheaper.

Reply:
same for the other occurrences

Reply:
Got it, thank you :)